core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -145,9 +148,16 @@ public FilterReduceExpressionsRule(Class<? extends Filter> filterClass,
       this(filterClass, true, relBuilderFactory);
     }
 
+    @Deprecated // to be removed before 2.0
     public FilterReduceExpressionsRule(Class<? extends Filter> filterClass,
         boolean matchNullability, RelBuilderFactory relBuilderFactory) {
-      super(filterClass, matchNullability, relBuilderFactory,
+      this(filterClass, newOptions().matchNullability(matchNullability).unknownAsFalse(true),
+          relBuilderFactory);
+    }
+
+    public FilterReduceExpressionsRule(Class<? extends Filter> filterClass,
+        ReductionOptions options, RelBuilderFactory relBuilderFactory) {
+      super(filterClass, options, relBuilderFactory,
```

Comment:
do you think it should assert that unknownAsFalse is set correctly when being provided options directly?